wagtailmodelchoosers/client/components/BaseChooser.js

```diff
@@ -20,14 +20,9 @@ const defaultProps = {
 const propTypes = {
   initialValue: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
   updateInputValue: PropTypes.func.isRequired,
-  label: PropTypes.string.isRequired,
   initial_display_value: PropTypes.string.isRequired,
-  endpoint: PropTypes.string.isRequired,
-  value: PropTypes.any,
   required: PropTypes.bool.isRequired,
   display: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
-  list_display: PropTypes.array.isRequired,
-  filters: PropTypes.array,
```

Comment:
Yeah I thought it would be a tricky case... To be honest I just wanted to make the linter happy.
I don't really know what's the state of art in that case...
I'm not a big fan of `{...this.props}` to be fair. Is there anything we can't list that could be passed to ModelPicker?
And that's kind of the problem with `...this.props` it's it allows basically everything. I read somewhere it wasn't really good practice. So I would be keen to decompose this.props into single vars we pass to the component.

What do you think?